nilearn/input_data/nifti_maps_masker.py

```diff
@@ -149,28 +190,11 @@ def fit(self, X=None, y=None):
 
         # Check shapes and affines or resample.
         if self.resampling_target is None and self.mask_img_ is not None:
-            if self.mask_img_.shape != self.maps_img_.shape[:3]:
-                raise ValueError(
-                    _compose_err_msg(
-                        "Regions and mask do not have the same shape",
-                        mask_img=self.mask_img, maps_img=self.maps_img))
-            if not np.allclose(self.mask_img_.get_affine(),
-                               self.maps_img_.get_affine()):
-                raise ValueError(
-                    _compose_err_msg(
-                        "Regions and mask do not have the same affine.",
-                        mask_img=self.mask_img, maps_img=self.maps_img))
-
-            # Since a copy is required, order can be forced as well.
-            maps_data = _utils.as_ndarray(self.maps_img_.get_data(),
-                                          copy=True, order="C")
-            maps_affine = _utils.as_ndarray(
-                                            self.maps_img_.get_affine())
-            self.maps_img_ = new_img_like(self.maps_img_, maps_data,
-                                          maps_affine)
+            _assert_same_fov(mask=self.mask_img_, maps=self.maps_img_)
 
         elif self.resampling_target == "mask" and self.mask_img_ is not None:
-            logger.log("resampling regions", verbose=self.verbose)
+            if self.verbose > 0:
```

Comment:
This is a PR to fix bugs so I took the occasion to fix other problems. I can remove it if it's necessary though.